services/synthesis_service.py

```
@@ -0,0 +1,224 @@
+    
+    def __init__(self, api_key: Optional[str] = None, model: str = "gpt-4o"):
+        """
+        Initialize the synthesis service.
+        
+        Args:
+            api_key: OpenAI API key (defaults to OPENAI_API_KEY env var)
+            model: LLM model to use
+        """
+        self.api_key = api_key or os.getenv("OPENAI_API_KEY")
+        if not self.api_key:
+            raise ValueError("OpenAI API key is required but not provided")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)